eng/common/pipelines/templates/steps/update-docsms-metadata.yml

```diff
@@ -26,7 +26,12 @@ parameters:
     type: object
     default:
       - '**'
-
+  - name: TargetVariable
```

Comment:
Upshot: Remove these parameters and use a name that's defined within the scope of the template

I don't think we would need to change the name of this variable when applying this template. We should remove these parameters and specify a static name for `TargetVariable` ... That way YAMLs that use this template don't need to specify `TargetVariable` or `TargetUserVariable`. 

If we merged this today we wouldn't be able to take advantage of this until we update every `archetype-*-release.yml` to include a name here.  

If this template were to be called several times across, for example, different repos, it might make sense to use parameters to set the variable names here. An example of that is instances of [`set-default-branch.yml`](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/pipelines/templates/steps/set-default-branch.yml) called multiple times in the same job for different repositories. It's not directly applicable in this context but if we could make a case then we could parameterize the variable names.